src/System.Private.CoreLib/shared/System/StringComparer.cs

```diff
@@ -367,7 +367,20 @@ public OrdinalIgnoreCaseComparer() : base(true)
 
         public override int Compare(string x, string y) => string.Compare(x, y, StringComparison.OrdinalIgnoreCase);
 
-        public override bool Equals(string x, string y) => string.Equals(x, y, StringComparison.OrdinalIgnoreCase);
+        public override bool Equals(string x, string y)
+        {
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            if (x is null || y is null)
+            {
+                return false;
+            }
+
+            return CompareInfo.EqualsOrdinalIgnoreCase(ref x.GetRawStringData(), ref y.GetRawStringData(), x.Length);
+        }
```

Comment:
This is effectively just manually inlining the previous string.Equals call?

Reply:
Generally, yeah. It's also skipping the unnecessary overhead that `String.Equals(...)` performs re: length checking, referential equality, and validation of `StringComparison` parameter, because the caller knows these checks are irrelevant.

Reply:
This is checking reference equality.

Why isn't the length check beneficial here?

For validation of the StringComparison, since it was a constant here I would have hoped if it actually was inlined (I realize it wouldn't be without nudging) the JIT would then be able to prune those checks away.

Reply:
Whoops, that's an oversight. It looks like the unit tests are passing regardless, so I'll add a test for this condition to check for future regressions.

Reply:
So at this point it's saving a non-virtual method call, a jump in a switch, and a comparison on the StringComparison enum value.  Given all the other work being done, does that end up being measurable?  If so, ok.  If not, it'd be better to avoid the code duplication.

Reply:
I'll rerun a test with only this change and get new numbers.

Reply:
|                                          Method |         Toolchain |     Mean |     Error |    StdDev | Scaled |
|------------------------------------------------ |------------------ |---------:|----------:|----------:|-------:|
| EqualsOrdinalIgnoreCase_StringsDifferOnlyInCase | modified comparer | 5.017 ms | 0.0637 ms | 0.0532 ms |   0.75 |
| EqualsOrdinalIgnoreCase_StringsDifferOnlyInCase | original comparer | 6.675 ms | 0.1057 ms | 0.0989 ms |   1.00 |

Reply:
I believe the perf boost is coming from a combination of not having to perform register shuffling (since the `StringComparer.Equals` signature is different from the `String.Equals` signature) and providing the ability for the JIT to tail-call into the workhorse method due to no shuffling being required with the modified comparer.